File: CarCareHub_/CarCareHub.Model/BPAutodijeloviAutoservis.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CarCareHub.Model
{
    public class BPAutodijeloviAutoservis
    {
        public int? BPAutodijeloviAutoservisId { get; set; }
        public virtual FirmaAutodijelova? FirmaAutodijelova { get; set; }
        public virtual Autoservis? Autoservis { get; set; }

        public int? FirmaAutodijelovaID { get; set; }
        public int? AutoservisId { get; set; }
        public bool Vidljivo { get; set; }
    }
}

File: CarCareHub_/UI_/flutter_mobile/lib/models/BPAutodijeloviAutoservis.dart
import 'package:flutter_mobile/models/autoservis.dart';
import 'package:flutter_mobile/models/firmaautodijelova.dart';
import 'package:flutter_mobile/models/grad.dart';
import 'package:json_annotation/json_annotation.dart';

part 'BPAutodijeloviAutoservis.g.dart';

@JsonSerializable()
class BPAutodijeloviAutoservis {
  int? bpAutodijeloviAutoservisId;
  FirmaAutodijelova? firmaAutodijelova;
  Autoservis? autoservis;
  int? firmaAutodijelovaId;
  bool? vidljivo;
  int? autoservisId;

  BPAutodijeloviAutoservis(
    this.bpAutodijeloviAutoservisId,
    this.vidljivo,
    this.firmaAutodijelova,
    this.firmaAutodijelovaId,
    this.autoservis,
    this.autoservisId,
  );

  factory BPAutodijeloviAutoservis.fromJson(Map<String, dynamic> json) =>
      _$BPAutodijeloviAutoservisFromJson(json);

  Map<String, dynamic> toJson() => _$BPAutodijeloviAutoservisToJson(this);
}
